src/core/thread/mle.cpp

```diff
@@ -788,15 +788,20 @@ ThreadError Mle::AppendLeaderData(Message &aMessage)
     return aMessage.Append(&mLeaderData, sizeof(mLeaderData));
 }
 
+void Mle::FillNetworkDataTlv(NetworkDataTlv &aTlv, bool aStableOnly)
```

Comment:
The type value for NetworkData is different between MLE TLV and NetworkDiagnostic TLV. Since AppendNetworkData() appends data directly to Message. There is no clean way to correct the Type value in the TLV.